### PR TITLE
Migrate Semeion prototype dataset

### PR DESCRIPTION
**Pull request by NicolasHug**

### CODE REVIEW

File: torchvision/prototype/datasets/_builtin/semeion.py [hunk 1]
```diff
@@ -36,18 +48,16 @@ def _prepare_sample(self, data: Tuple[str, ...]) -> Dict[str, Any]:
         image_data, label_data = data[:256], data[256:-1]
 
         return dict(
-            image=Image(torch.tensor([float(pixel) for pixel in image_data], dtype=torch.uint8).reshape(16, 16)),
-            label=OneHotLabel([int(label) for label in label_data], categories=self.categories),
+            image=Image(torch.tensor([float(pixel) for pixel in image_data], dtype=torch.float).reshape(16, 16)),
```
Comment (NicolasHug) [inline torchvision/prototype/datasets/_builtin/semeion.py:51]: @pmeier  the original PR used `uint8` here, but these are float values in [0, 1]. Do you know if the use of `dtype=uint8` was inteded for something specific?


Also, the manual conversion of every single value is fairly inefficient for all of the 16 * 16 pixels. Do you know if there's a more efficient conversion that could be applied at the CSVParser level?

Comment (pmeier) [reply torchvision/prototype/datasets/_builtin/semeion.py:51]: > the original PR used `uint8` here, but these are float values in [0, 1]. Do you know if the use of `dtype=uint8` was inteded for something specific?

Nope, that seems like a bug. Good catch!

> Also, the manual conversion of every single value is fairly inefficient for all of the 16 * 16 pixels. Do you know if there's a more efficient conversion that could be applied at the CSVParser level?

I'm not aware of anything. Without a user request for a better performance, I wouldn't sweat it though. I'm guessing this dataset is not in widespread use and the code will probably get more complicated. If you have a simple solution, go for it.

Comment (pmeier) [reply torchvision/prototype/datasets/_builtin/semeion.py:51]: Just saw that this dataset has only 1.5k samples. If someone wants more speed, the best way out is probably to just load the whole dataset into memory.

**Review by pmeier — APPROVED** [review]
LGTM, thanks Nicolas!

**Pull request merged**